pkg/cmd/registry/artifact/state/state.go

```diff
@@ -0,0 +1,121 @@
+	Connection factory.ConnectionFunc
+	localizer  localize.Localizer
+}
+
+// NewSetMetadataCommand creates a new command for updating metadata for registry artifacts.
+func NewSetStateCommand(f *factory.Factory) *cobra.Command {
+	opts := &SetOptions{
+		Config:     f.Config,
+		Connection: f.Connection,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Logger:     f.Logger,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "state-set",
+		Short:   f.Localizer.MustLocalize("artifact.cmd.stateset.description.short"),
+		Long:    f.Localizer.MustLocalize("artifact.cmd.stateset.description.long"),
+		Example: f.Localizer.MustLocalize("artifact.cmd.stateset.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.artifact == "" {
+				return errors.New(f.Localizer.MustLocalize("artifact.common.message.artifactIdRequired"))
+			}
+
+			if opts.state != "" {
+				if _, err := registryinstanceclient.NewArtifactStateFromValue(opts.state); err != nil {
+					return errors.New(opts.localizer.MustLocalize("artifact.cmd.state.error.invalidArtifactState", localize.NewEntry("AllowedTypes", util.GetAllowedArtifactStateEnumValuesAsString())))
+				}
+			}
+
+			if opts.registryID != "" {
+				return runSet(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return errors.New(opts.localizer.MustLocalize("registry.no.service.selected.use.instance.id.flag"))
+			}
+
+			opts.registryID = cfg.Services.ServiceRegistry.InstanceID
+			return runSet(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.artifact, "artifact-id", "a", "", opts.localizer.MustLocalize("artifact.common.id"))
+	cmd.Flags().StringVarP(&opts.group, "group", "g", util.DefaultArtifactGroup, opts.localizer.MustLocalize("artifact.common.group"))
+	cmd.Flags().StringVar(&opts.registryID, "instance-id", "", opts.localizer.MustLocalize("artifact.common.instance.id"))
+	cmd.Flags().StringVar(&opts.state, "state", "", opts.localizer.MustLocalize("artifact.flag.state.description"))
+
+	_ = cmd.RegisterFlagCompletionFunc("state", func(cmd *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
+		return util.AllowedArtifactStateEnumValues, cobra.ShellCompDirectiveNoSpace
+	})
+	return cmd
+}
+
+func runSet(opts *SetOptions) error {
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	dataAPI, _, err := conn.API().ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+
+	if opts.group == util.DefaultArtifactGroup {
+		opts.Logger.Info(opts.localizer.MustLocalize("artifact.common.message.no.group", localize.NewEntry("DefaultArtifactGroup", util.DefaultArtifactGroup)))
+		opts.group = util.DefaultArtifactGroup
+	}
+
+	updateState, err := registryinstanceclient.NewArtifactStateFromValue(opts.state)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
```

Comment:
Don't you want to use the context from the factory object? :-D

Reply:
That wasn't rebased yet.

Reply:
I know.